api/users.go

```diff
@@ -112,6 +112,60 @@ func (api *API) GetUserHandler(ctx context.Context, w http.ResponseWriter, req *
 	return models.NewSuccessResponse(jsonResponse, http.StatusOK, nil), nil
 }
 
+//UpdateUserHandler updates a users details in Cognito and returns a http handler interface
+func (api *API) UpdateUserHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	defer req.Body.Close()
+	vars := mux.Vars(req)
+
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, handleBodyReadError(ctx, err)
+	}
+
+	user := models.UserParams{}
+	err = json.Unmarshal(body, &user)
+	if err != nil {
+		return nil, handleBodyUnmarshalError(ctx, err)
+	}
+	user.ID = vars["id"]
+
+	validationErrs := user.ValidateUpdate(ctx)
+
+	if len(validationErrs) != 0 {
+		return nil, models.NewErrorResponse(validationErrs, http.StatusBadRequest, nil)
+	}
+
+	userRequest := user.BuildUpdateUserRequest(api.UserPoolId)
+
+	_, err = api.CognitoClient.AdminUpdateUserAttributes(userRequest)
+	if err != nil {
+		responseErr := models.NewCognitoError(ctx, err, "AdminUpdateUserAttributes request from update user endpoint")
+		if responseErr.Code == models.UserNotFoundError {
+			return nil, models.NewErrorResponse([]error{responseErr}, http.StatusNotFound, nil)
+		} else if responseErr.Code == models.InvalidFieldError {
+			return nil, models.NewErrorResponse([]error{responseErr}, http.StatusBadRequest, nil)
+		} else {
+			return nil, models.NewErrorResponse([]error{responseErr}, http.StatusInternalServerError, nil)
+		}
+	}
+
+	userDetailsRequest := user.BuildAdminGetUserRequest(api.UserPoolId)
+	userDetailsResponse, err := api.CognitoClient.AdminGetUser(userDetailsRequest)
+	if err != nil {
+		responseErr := models.NewCognitoError(ctx, err, "AdminGetUser request from update user endpoint")
+		return nil, models.NewErrorResponse([]error{responseErr}, http.StatusInternalServerError, nil)
```

Comment:
This is probably beyond the scope of this PR but I am thinking adding a new func to make this syntax a bit nice might be a good idea:
Atm we have:
```
func NewErrorResponse(errors []error, statusCode int, headers map[string]string) *ErrorResponse {}
```
But that requires you to do the following all the time when you only have 1 error.
```
[]error{responseErr}
```
Its driving my OCD wild 😂  We could add another "overloaded func":
```
func NewErrorResponse(statusCode int, headers map[string]string, errors... error) *ErrorResponse {}
```
Which would be a bit nicer to use. Again totally beyond the scope of this PR.

Reply:
This has actually been bugging me too.
The suggested approach is definitely cleaner.
I'll open a separate PR to refactor this across the service